superset-frontend/src/components/Chart/Chart.jsx

```diff
@@ -232,16 +233,68 @@ class Chart extends React.PureComponent {
     );
   }
 
+  renderSpinner(chartStatus, databaseName) {
+    const messages = {
+      loading: t('Fetching data from %s', databaseName),
+      success: t('Rendering chart'),
+      rendered: t('Success'),
+    };
+
+    return (
+      <div
+        css={css`
+          position: absolute;
+          left: 50%;
+          top: 50%;
+          transform: translate(-50%, -50%);
+        `}
+      >
+        <Loading position="inline-centered" />
+        <span
+          // eslint-disable-next-line theme-colors/no-literal-colors
+          css={css`
+            display: block;
+            margin: 16px auto;
+            width: fit-content;
+            color: #666666;
```

Comment:
```suggestion

    const theme = useTheme();

    return (
      <div
        css={css`
          position: absolute;
          left: 50%;
          top: 50%;
          transform: translate(-50%, -50%);
        `}
      >
        <Loading position="inline-centered" />
        <span
          css={css`
            display: block;
            margin: 16px auto;
            width: fit-content;
            color: theme.colors.grayscale.base;
```

Reply:
You would also need to `import { useTheme } from '@superset-ui/core';`

Reply:
Thanks!